lua/orgmode/colors/highlighter/stars.lua

```diff
@@ -15,7 +15,7 @@ function OrgStars:new(opts)
 end
 
 function OrgStars:on_line(bufnr, line)
-  if not config.org_hide_leading_stars then
+  if not config.org_hide_leading_stars or not (vim.b.org_indent_mode and config.org_hide_leading_stars) then
```

Comment:
Lets use `org_indent_mode` + `org_indent_mode_turns_on_hiding_stars` here. Since it's a more complex check, you can extract it to a private method `_is_enabled`:

```lua
function OrgStars:_is_enabled(bufnr)
  if config.org_hide_leading_stars then
    return true
  end

  if vim.b[bufnr].org_indent_mode and config.org_indent_mode_turns_on_hiding_stars then
    return true
  end

  return false
end
```

Reply:
Done as part of latest force push.